### DISCUSSION TITLE
Snippets broken in 1.107

### DISCUSSION THREAD

**Discussion by croaker000** [pulsar-edit/.github#203] [Pulsar Support] [answered]
Extract from my snippets.cson file

```
'.source.php':
   'Debug_Query':
     'prefix': 'debug_query'
     'body': 'echo \'<pre>DEBUG QUERY:: \'.print_r($this->MODEL->debug_query(false,false,true),true).\'</pre>\';'
```

In 1.106, it would return a portion as "$this->MODEL->debug_query". In 1.107, I get "->MODEL->debug_query" (the "$this" bit disappears)

Should I be escaping the "$" now or is this a bug related to the new less upgrade?

**Comment by savetheclocktower** [comment]
It's not related to the Less upgrade; it'd be related to the new support for variables in the Snippets package.

Please do try to escape the `$` and let me know if that works. I'll add a test for this. Thanks for reporting it!

**Reply by savetheclocktower** [reply, marked as answer]
OK, I've verified that this works as long as you double-escape the `$`:

```coffeescript
'.source.php':
   'Debug_Query':
     'prefix': 'debug_query'
     'body': 'echo \'<pre>DEBUG QUERY:: \'.print_r(\\$this->MODEL->debug_query(false,false,true),true).\'</pre>\';'
```

This is happening because the snippet body parser now recognizes `$this` as a variable just like, say, `$TM_CURRENT_LINE`. The LSP spec for snippets explicitly says that `$`s need to be escaped with backslashes. And the _double_ escape is necessary because CSON string parsing uses backslash escapes, so it needs to treat that as a literal backslash so that it survives until the snippets package can consume it.

I regret not realizing that this would be a new constraint — I don't like breaking existing snippets if I don't have to. But this was ultimately going to be necessary for compatibility with VSCode and other snippet implementations. If this affects enough people, I'll look into writing something that tries to detect these scenarios and transforms the snippet bodies before they get parsed.

**Reply by croaker000** [reply]
Thanks for the clarification. Sorry about the delay in replying - timezones!
